### DISCUSSION TITLE
`enum.nonmember` and `enum.member` Stubs

### DISCUSSION THREAD

**Discussion by max-muoto** [microsoft/pyright#7950] [General]
It seems the typeshed maintainers want to ensures the stubs for `enum.nonmember` or `enum.member` are consistent with the cpython implementations: https://github.com/python/typeshed/issues/11921, despite different runtime behavior.

This causes issues in Pyright, as attributes such as `.value`, might not be available, though the stubs indicate they are:

```python
import enum


class MyEnum(enum.StrEnum):
    VAL_1 = "val_1"
    VAL_2 = "val_2"

    _items = enum.nonmember({VAL_1: 1, VAL_2: 2})

    def to_int(self) -> int:
        # Would pass Pyright
        return self._items.value[self]

#  'dict' object has no attribute 'value'
MyEnum.VAL_1.to_int()
```

[Pyright Playground
](https://pyright-play.net/?code=JYWwDg9gTgLgBAUwHYFcQChMGMA2BDAZwLgFkBPAUVRAApk0A6AZRiirQEoAudOPuAGoBBADIB9AIxwAvHABEANzw5Jc3v2HiATDPlKVWter5jgMBCGKz6IBkghIQFgEYIoNAN6bJXOBIA0gqJiWr5aAL4cmPxwACYIAGZwMBCmSDA0BAg4CRxwALQAfHDA6TwxMQDEcADqECg4sXBghMQACmRQwADmABYwxhVQCDAoUEhwWTkMpuaWDPooCADaUwkAupjVcADkscBYMDtwEM4AVgiHcL2EcPZweDCswM4o5ruLCDvo5Oy23hIGCk0hkokA)
Curious if there any thoughts about avoiding reliance on these stubs, and just implementing custom support?

**Comment by erictraut** [comment]
OK, thanks for tracking down those details. I've re-opened the original bug report that you filed on this issue (https://github.com/microsoft/pyright/issues/7924).

**Reply by max-muoto** [reply]
Thanks Eric, really appreciate it!
